runtime/android/core_internal/src/org/xwalk/core/internal/XWalkUIClientInternal.java

```diff
@@ -522,4 +531,100 @@ public void onCancel(DialogInterface dialog) {
         mDialog.show();
         return false;
     }
+
+    void setContentsClient(XWalkContentsClient client) {
+        mContentsClient = client;
+    }
+
+    private Activity addContentView(View view, CustomViewCallbackInternal callback) {
+        Activity activity = mXWalkView.getActivity();
+        if (mCustomXWalkView != null || activity == null) {
+            if (callback != null) callback.onCustomViewHidden();
```

Comment:
Before adding custom view, we should check the condition.
The callback was exposed in `CustomViewCallbackInternal.java` for user to invoke.